source/docs/software/actuators/programming-leds.rst

```diff
@@ -0,0 +1,147 @@
+Programming LEDs
+================
+
+LED strips have been commonly used by teams for several years for a variety of reasons. They allow teams to debug robot functionality from the audience, provide a visual marker for their robot, and can simply add some visual appeal. WPILib has an API for controlling WS2812 LEDs with their data pin connected via PWM.
+
+Instantiating the AddressableLED Object
+---------------------------------------
+
+You first create an ``AddressableLED`` object that takes the PWM port as an argument. It *must* be a PWM header on the roboRIO. Then you set the number of LEDs located on your LED strip, with can be done with the ``setLength()`` function.
+
+.. important:: It is important to note that setting the length of the LED header is an expensive task and it's **not** recommended to run this periodically.
+
+After the length of the strip has been set, you'll have to create an ``AddressableLEDBuffer`` object that takes the number of LEDs as an input. You'll then call ``myAddressableLed.setData(myAddressableLEDBuffer)`` to set the led output data. Finally, you can call ``myAddressableLed.start()`` to write the output continuously. Below is a full example of the initialization process.
+
+.. note:: C++ does not have an AddressableLEDBuffer, and instead uses an Array.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
```

Comment:
This doesn't show all of the relevant code. Stops at the beginning of robot init